### PR TITLE
tools/dm: fix links

### PR DESCRIPTION
**Pull request by dcalvin**
fixed a few dead links

### CODE REVIEW

**Comment by dcalvin** [comment]
@lilin90 PTLA

File: tools/dm/deployment.md [hunk 1]
```diff
@@ -289,7 +289,7 @@ grafana_admin_password = "admin"
 | server_id | DM-worker 伪装成一个 MySQL slave，该变量即为这个 slave 的 server ID，在 MySQL 集群中需保持全局唯一。取值范围 0 ~ 4294967295。|
 | mysql_host | 上游 MySQL 主机 |
 | mysql_user | 上游 MySQL 用户名，默认值为 “root”。|
-| mysql_password | 上游 MySQL 用户密码，需使用 `dmctl` 工具加密。请参考 [使用 dmctl 加密上游 MySQL 用户密码](#encrypt-the-upstream-mysql-user-password-using-dmctl). |
+| mysql_password | 上游 MySQL 用户密码，需使用 `dmctl` 工具加密。请参考[使用 dmctl 加密上游 MySQL 用户密码](#使用-dmctl-加密-mysql-用户密码). |
```
Comment (lilin90) [inline tools/dm/deployment.md]:
```suggestion
| mysql_password | 上游 MySQL 用户密码，需使用 `dmctl` 工具加密。请参考[使用 dmctl 加密上游 MySQL 用户密码](#使用-dmctl-加密上游-mysql-用户密码)。 |
```

File: tools/dm/deployment.md [hunk 2]
```diff
@@ -183,11 +183,11 @@ dm_worker1 ansible_host=172.16.10.72 server_id=101 source_id="mysql-replica-01"
 
 根据场景需要，您可以在以下两种集群拓扑中任选一种：
 
-- [单节点上单个 DM-worker 实例的集群拓扑](#选项-1-使用单节点单个-DM-Worker-实例的集群拓扑)
+- [单节点上单个 DM-worker 实例的集群拓扑](#选项-1使用单节点上单个-dm-worker-实例的集群拓扑)
 
-- [单节点上多个 DM-worker 实例的集群拓扑](#选项-2-使用单节点多个-DM-Worker-实例的集群拓扑)
+- [单节点上多个 DM-worker 实例的集群拓扑](#选项-2使用单节点多个-dm-worker-实例的集群拓扑)
```
Comment (lilin90) [inline tools/dm/deployment.md]:
```suggestion
- [单节点上多个 DM-worker 实例的集群拓扑](#选项-2使用单节点上多个-dm-worker-实例的集群拓扑)
```

File: tools/dm/deployment.md [hunk 3]
```diff
@@ -183,11 +183,11 @@ dm_worker1 ansible_host=172.16.10.72 server_id=101 source_id="mysql-replica-01"
...
 
-通常情况下，我们推荐每个节点上部署单个 DM-Worker 实例。但如果您的机器拥有性能远超 [TiDB 软件和硬件环境要求](/op-guide/recommendation.md)中推荐配置的CPU和内存，并且每个节点配置 2 块以上的硬盘或大于 2T 的 SSD，您可以在单个节点上部署不超过 2 个 DM-Worker 实例。
+通常情况下，我们推荐每个节点上部署单个 DM-Worker 实例。但如果您的机器拥有性能远超[TiDB 软件和硬件环境要求](/op-guide/recommendation.md)中推荐配置的CPU和内存，并且每个节点配置 2 块以上的硬盘或大于 2T 的 SSD，您可以在单个节点上部署不超过 2 个 DM-Worker 实例。
```
Comment (lilin90) [inline tools/dm/deployment.md]:
```suggestion
通常情况下，我们推荐每个节点上部署单个 DM-Worker 实例。但如果您的机器拥有性能远超 [TiDB 软件和硬件环境要求](/op-guide/recommendation.md)中推荐配置的CPU和内存，并且每个节点配置 2 块以上的硬盘或大于 2T 的 SSD，您可以在单个节点上部署不超过 2 个 DM-Worker 实例。
```

**Pull request merged**